File: backend/app/normalizers/page.py
from .section import normalize_section

def normalize_page(page, admin=False, preview=False):
    sections = sorted(page.sections, key=lambda s: s.order)

    return {
        "id": page.id,
        "title": page.title,
        "slug": page.slug,
        "status": page.status if admin else None,
        "seo": page.seo or {},
        "sections": [
            normalize_section(
                s,
                admin=admin,
                include_blocks=True
            )
            for s in sections
        ]
    }

File: frontend/shared/types/page.ts
export type PageStatus = "draft" | "published"
import type { Block } from "./block"

export type PageSummary = {
  id: string
  title: string
  slug: string
  status: PageStatus
  updated_at: string // ISO string from backend
}

/**
 * Full page payload returned by:
 * GET /v1/pages/:id
 *
 * Mirrors backend normalizer output.
 */
export type PageDetail = {
  id: string
  title: string
  slug: string
  status: PageStatus
  version: number
  sections: PageSection[]
  updated_at: string
  published_at: string | null
}

export type PageSection = {
  id: string
  title: string
  order: number
  blocks: Block[]
}
